### PR TITLE
ENH: Add 'keepdims' to 'average()' and 'ma.average()'.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
I was surprised to find that this was not already implemented.

There are a few trivial PEP 8 blank line adjustments included.  I hope they are not too distracting.

### CODE REVIEW

**Comment by mattip** [comment]
Looks good to me. A release note would be nice.

**Review by seberg — APPROVED** [review]
Also LGTM, as Matti just posted, the release note may be good, but otherwise I think we can get this in for the upcoming release.

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -524,7 +547,8 @@ def average(a, axis=None, weights=None, returned=False):
             raise ZeroDivisionError(
                 "Weights sum to zero, can't be normalized")
 
-        avg = np.multiply(a, wgt, dtype=result_dtype).sum(axis)/scl
+        avg = np.multiply(a, wgt,
+                          dtype=result_dtype).sum(axis, **keepdims_kw) / scl
```
[the remark this answers is not in the document]
Comment (seberg) [reply numpy/lib/function_base.py:551]: Ah, for the `arr.mean()` path, you would need:
```
if keepdims is np._NoValue:
    avg = arr.mean()
else:
    avg = arr.mean(keepdims=keepdims)
```
but yeah, it requires two `if`s and it really doesn't matter much, if anyone ever cares one way or another we can follow up.  For now, just waiting for tests to finish.

**Comment by WarrenWeckesser** [comment]
I forgot to update the typing information in `function_base.pyi`.  I'll create a follow-up PR to fix that.

**Comment by WarrenWeckesser** [comment]
Thanks @BvB93, pull request is https://github.com/numpy/numpy/pull/21490

**Pull request merged**